include/modbus/modbus_rtu.h

```diff
@@ -0,0 +1,396 @@
+ *                   the server.  The array pointed to by 'reg_buf' needs
+ *                   to be able to hold at least 'num_regs' entries.
+ * @param num_regs   Quantity of registers to read
+ *
+ * @retval           0 If the function was successful
+ */
+int mb_rtu_read_holding_regs_fp(const int iface,
+				const uint8_t node_addr,
+				const uint16_t start_addr,
+				float *const reg_buf,
+				const uint16_t num_regs);
+
+/**
+ * @brief Write floating-point holding registers (FC16)
+ *
+ * Sends a Modbus message to write to floating-point holding registers
+ * to a server unit.
+ *
+ * @param iface      Modbus RTU interface index
+ * @param node_addr  Modbus RTU address of the server
+ * @param start_addr Register starting address
+ * @param reg_buf    Is a pointer to an array containing
+ *                   the value of the holding registers to write.
+ *                   Note that the array containing the register values must
+ *                   be greater than or equal to 'num_regs'
+ * @param num_regs   Quantity of registers to write
+ *
+ * @retval           0 If the function was successful
+ */
+int mb_rtu_write_holding_regs_fp(const int iface,
+				 const uint8_t node_addr,
+				 const uint16_t start_addr,
+				 float *const reg_buf,
+				 const uint16_t num_regs);
+
+/** Modbus Server User Callback structure */
+struct mbs_rtu_user_callbacks {
+	/** Coil read callback */
+	int (*coil_rd)(uint16_t addr, bool *state);
+
+	/** Coil write callback */
+	int (*coil_wr)(uint16_t addr, bool state);
+
+	/** Discrete Input read callback */
+	int (*discrete_input_rd)(uint16_t addr, bool *state);
+
+	/** Input Register read callback */
+	int (*input_reg_rd)(uint16_t addr, uint16_t *reg);
+
+	/** Floating Point Input Register read callback */
+	int (*input_reg_rd_fp)(uint16_t addr, float *reg);
+
+	/** Holding Register read callback */
+	int (*holding_reg_rd)(uint16_t addr, uint16_t *reg);
+
+	/** Holding Register write callback */
+	int (*holding_reg_wr)(uint16_t addr, uint16_t reg);
+
+	/** Floating Point Holding Register read callback */
+	int (*holding_reg_rd_fp)(uint16_t addr, float *reg);
+
+	/** Floating Point Holding Register write callback */
+	int (*holding_reg_wr_fp)(uint16_t addr, float reg);
+};
+
+/**
+ * @brief Configure Modbus Interface as server
+ *
+ * @param iface      Modbus RTU interface index
+ * @param node_addr  Modbus RTU address of the server
+ * @param baud       Baudrate of the serial line
+ * @param parity     UART's parity setting:
+ *                       UART_CFG_PARITY_NONE,
+ *                       UART_CFG_PARITY_EVEN,
+ *                       UART_CFG_PARITY_ODD
+ * @param cb         Pointer to the User Callback structure
+ * @param ascii_mode Enable ASCII Transfer Mode
+ *
+ * @retval           0 If the function was successful
+ */
+int mb_rtu_cfg_server(const uint8_t iface, const uint8_t node_addr,
+		      const uint32_t baud, enum uart_config_parity parity,
```

Comment:
Transport specific configurations could be encapsulated in a `struct modbus_cfg` and provided as a pointer to the struct - again, for the sake of possible future extension with TCP mode.